### ISSUE TITLE
RDKitConverter fails to infer bond orders/charges correctly

### ISSUE THREAD

**Issue by cbouy** [MDAnalysis/mdanalysis#3339]
## Expected behavior ##

The RDKitConverter should be able to infer correctly bond orders and formal charges for any molecule

## Actual behavior ##

As mentioned in #3044, it currently fails for molecules with the following patterns:
* `R=[N+](-[O-])-R`
* `R-[N+]#N`
* `R-N=[N+]=[N-]`
* `R-[O+]=R`
* `R-[N+]#[C-]`
* `R-[N-]-R`

## Code to reproduce the behavior ##

See the RDKitConverter benchmark for examples of molecules *(it's on my TODO list for now...)*

## Current version of MDAnalysis ##

- Which version are you using? `2.0.0b0`

**Comment by orbeckst** [comment]
Will the issue be fixed by PR #3044 ?

**Comment by cbouy** [comment]
Nope! This issue keeps track of the cases that are known to fail even after PR#3044.
